spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-schedulerx/src/main/java/com/alibaba/cloud/scheduling/schedulerx/service/JobSyncService.java

```diff
@@ -0,0 +1,434 @@
+			request.setMonitorContactsJson(JsonUtil.toJson(contactInfos));
+		}
+		CreateAppGroupResponse response = client.getAcsResponse(request);
+		if (response.getSuccess()) {
+			LOGGER.info(JsonUtil.toJson(response));
+			return true;
+		}
+		else {
+			throw new IOException(response.getMessage());
+		}
+	}
+
+	/**
+	 * Get job config info.
+	 *
+	 * @param client pop client
+	 * @param jobName job name
+	 * @param namespaceSource namespace source
+	 * @return job config info.
+	 * @throws Exception get job config info exception.
+	 */
+	private JobConfigInfo getJob(DefaultAcsClient client, String jobName, String namespaceSource) throws Exception {
+		GetJobInfoRequest request = new GetJobInfoRequest();
+		request.setNamespace(properties.getNamespace());
+		request.setNamespaceSource(namespaceSource);
+		request.setGroupId(properties.getGroupId());
+		request.setJobId(0L);
+		request.setJobName(jobName);
+		GetJobInfoResponse response = client.getAcsResponse(request);
+		if (response.getSuccess()) {
+			return response.getData().getJobConfigInfo();
+		}
+		return null;
+	}
+
+	/**
+	 * create job.
+	 *
+	 * @param client pop client
+	 * @param jobName job name
+	 * @param jobProperty job property
+	 * @param namespaceSource namespace source
+	 * @throws Exception create job exception
+	 */
+	private void createJob(DefaultAcsClient client, String jobName, JobProperty jobProperty, String namespaceSource) throws Exception {
+		CreateJobRequest request = new CreateJobRequest();
+		request.setNamespace(properties.getNamespace());
+		request.setNamespaceSource(namespaceSource);
+		request.setGroupId(properties.getGroupId());
+		request.setName(jobName);
+		request.setParameters(jobProperty.getJobParameter());
+
+		if (jobProperty.getJobType().equals(JobType.JAVA.getKey())) {
+			request.setJobType("java");
+			request.setClassName(jobProperty.getClassName());
+		}
+		else {
+			request.setJobType(jobProperty.getJobType());
+		}
+
+		if (jobProperty.getJobModel().equals("mapreduce")) {
+			request.setExecuteMode("batch");
+		}
+		else {
+			request.setExecuteMode(jobProperty.getJobModel());
+		}
+		if (StringUtils.isNotEmpty(jobProperty.getDescription())) {
+			request.setDescription(jobProperty.getDescription());
+		}
+
+		if (StringUtils.isNotEmpty(jobProperty.getContent())) {
+			request.setContent(jobProperty.getContent());
+		}
+
+		if (StringUtils.isNotEmpty(jobProperty.getCron()) && StringUtils.isNotEmpty(jobProperty.getOneTime())) {
+			throw new IOException("cron and oneTime shouldn't set together");
+		}
+		if (StringUtils.isNotEmpty(jobProperty.getCron())) {
+			CronExpression cronExpression = new CronExpression(jobProperty.getCron());
+			Date now = new Date();
+			Date nextData = cronExpression.getTimeAfter(now);
+			Date next2Data = cronExpression.getTimeAfter(nextData);
+			if (nextData != null && next2Data != null) {
+				long interval = (next2Data.getTime() - nextData.getTime()) / 1000;
+				if (interval < 60) {
```

Comment:
Maybe we can avoid using these magic values

Reply:
done